frontend/src/hooks/query/use-settings.ts

```diff
@@ -54,13 +54,11 @@ export const useSettings = () => {
   React.useEffect(() => {
     if (query.data?.PROVIDER_TOKENS_SET) {
       const providers = query.data.PROVIDER_TOKENS_SET;
-      const setProviders = (
-        Object.keys(providers) as Array<keyof typeof providers>
-      ).filter((key) => providers[key]);
+      const setProviders = Object.keys(providers) as Array<
```

Comment:
This changes to FE logic where we expect the BE to only send providers which are currently set (as opposed to every provider tagged with a boolean value)